frontend/integration-tests/tests/kubevirt/non-admin.scenario.ts

```diff
@@ -0,0 +1,140 @@
+import { execSync } from 'child_process';
+
+describe('Test nonadmin user behaviour', () => {
+  const nonAdminNS = `${testName}-nonadmin`;
+  const userRole = ['view', 'edit'];
+
+  const verifyPermissions = async function( perm: string, ns: string) {
+    await browser.get(`${appHost}/k8s/ns/${ns}/virtualmachines`);
+    if (perm === 'noAccess') {
+      // crudView.isLoaded doesn't work here.
+      await browser.sleep(6000);
+      expect(crudView.errorMessage.getText()).toContain('cannot list resource');
+
+      // TODO: check it can't use vm dialog once BZ1728523 is fixed.
+    }
+    if (perm === 'view') {
+      await crudView.isLoaded(); // no error indicates it can view resource.
+      // TODO: check it can't use vm dialog once BZ1728523 is fixed.
+    }
+    if (perm === 'edit') {
+      await crudView.isLoaded();
+      await click(crudView.createItemButton);
+      await click(createWithWizardLink);
+      await crudView.isLoaded();
+      await click(closeWizard);
+
+      await click(crudView.createItemButton);
+      await click(createWithYAMLLink);
+      await yamlView.isLoaded();
+    }
+
+    await browser.get(`${appHost}/k8s/ns/${ns}/vmtemplates`);
+    if (perm === 'noAccess') {
+      await browser.sleep(6000);
+      expect(crudView.errorMessage.getText()).toContain('cannot list resource');
+    }
+    if (perm === 'view') {
+      await crudView.isLoaded(); // no error indicates it can view resource.
+      // TODO: check it can't use vm dialog once BZ1728523 is fixed.
+    }
+    if (perm === 'edit') {
+      await crudView.isLoaded();
+      await click(crudView.createItemButton);
+      await click(createWithWizardLink);
+      await crudView.isLoaded();
+      await click(closeWizard);
+    }
+
+    await browser.get(`${appHost}/overview/ns/${ns}/`);
+    if (perm === 'noAccess') {
+      await browser.sleep(6000);
+      expect(crudView.errorMessage.getText()).toContain('cannot list resource');
+    }
+    if (perm === 'view') {
+      await crudView.isLoaded(); // no error indicates it can view resource.
+      // TODO: check it can't use vm dialog once BZ1728523 is fixed.
+    }
+    if (perm === 'edit') {
+      await crudView.isLoaded();
+    }
+  };
+
+  const verifyPermissionWithRoleBinding = async function( perm: string, ns: string, roleBinding: string ) {
+    execSync(`oc adm policy add-${roleBinding}-to-user ${perm} ${BRIDGE_HTPASSWD_USERNAME} -n ${ns}`);
+    await verifyPermissions(perm, ns);
+    execSync(`oc adm policy remove-${roleBinding}-from-user ${perm} ${BRIDGE_HTPASSWD_USERNAME} -n ${ns}`);
+  };
+
+  beforeAll(async() => {
+    try {
+      execSync('oc get -o yaml -n openshift-config secret test');
+    } catch (error) {
+      createResources([nonAdminSecret, nonAdminProvider]);
+    }
+  });
+
+  afterAll(async() => {
+    await logout();
+    await browser.wait(until.presenceOf($('.login-pf')));
+    await loginView.login(KUBEADMIN_IDP, KUBEADMIN_USERNAME, BRIDGE_KUBEADMIN_PASSWORD);
```

Comment:
why are we doing this? Are other tests affected by this?

Reply:
It should affect later tests, so login with kubeadmin user after this test.